From a6860a943ef53df210775e9b4cd9412c5a7a333a Mon Sep 17 00:00:00 2001
From: Pauli <ppzgs1@gmail.com>
Date: Fri, 30 Aug 2024 11:43:29 +1000
Subject: [PATCH 222/228] endecode_test.c: Fix !fips v3.0.0 check

The fips_provider_version_* functions return true if the FIPS provider isn't
loaded.  This is somewhat counterintuitive and the fix in #25327 neglected
this nuance resulting in not running the SM2 tests when the FIPS provider
wasn't being loaded.

Reviewed-by: Viktor Dukhovni <viktor@openssl.org>
Reviewed-by: Tom Cosgrove <tom.cosgrove@arm.com>
Reviewed-by: Tomas Mraz <tomas@openssl.org>
(Merged from https://github.com/openssl/openssl/pull/25331)

(cherry picked from commit c6c6af18ea5f8dd7aa2bd54b63fcb813ee6c2394)
---
 test/endecode_test.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/test/endecode_test.c b/test/endecode_test.c
index 8e7dc7a9ec..bd28870a1b 100644
--- a/test/endecode_test.c
+++ b/test/endecode_test.c
@@ -1341,9 +1341,7 @@ int setup_tests(void)
     }
 
     /* FIPS(3.0.0): provider imports explicit params but they won't work #17998 */
-    is_fips_3_0_0 = fips_provider_version_eq(testctx, 3, 0, 0);
-    if (is_fips_3_0_0 < 0)
-        return 0;
+    is_fips_3_0_0 = is_fips && fips_provider_version_eq(testctx, 3, 0, 0);
 
 #ifdef STATIC_LEGACY
     /*
-- 
2.39.5 (Apple Git-154)

